tests/scalers_go/pulsar/pulsar_test.go

```diff
@@ -0,0 +1,258 @@
+//go:build e2e
+// +build e2e
+
+package pulsar_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "pulsar-test"
+)
+
+var (
+	testNamespace          = fmt.Sprintf("%s-ns", testName)
+	statefulSetName        = fmt.Sprintf("%s-sts", testName)
+	scaledObjectName       = fmt.Sprintf("%s-so", testName)
+	consumerDeploymentName = fmt.Sprintf("%s-consumer-deploy", testName)
+	producerJobName        = fmt.Sprintf("%s-producer-job", testName)
+	messageCount           = 3
+	minReplicaCount        = 0
+	maxReplicaCount        = 5
+	msgBacklog             = 10
+)
+
+type templateData struct {
+	TestNamespace          string
+	StatefulSetName        string
+	MessageCount           int
+	ScaledObjectName       string
+	ConsumerDeploymentName string
+	ProducerJobName        string
+	MinReplicaCount        int
+	MaxReplicaCount        int
+	MsgBacklog             int
+}
+
+const statefulsetTemplate = `
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+ name: {{.StatefulSetName}}
+ namespace: {{.TestNamespace}}
+ labels:
+  app: pulsar
+spec:
+  selector:
+    matchLabels:
+      app: pulsar
+  replicas: 1
+  serviceName: {{.StatefulSetName}}
+  template:
+    metadata:
+      labels:
+        app: pulsar
+    spec:
+      containers:
+      - name: pulsar
+        image: apachepulsar/pulsar:2.10.0
+        imagePullPolicy: IfNotPresent
+        readinessProbe:
+          tcpSocket:
+            port: 8080
+        ports:
+        - name: pulsar
+          containerPort: 6650
+          protocol: TCP
+        - name: admin
+          containerPort: 8080
+          protocol: TCP
+        env:
+        - name: PULSAR_PREFIX_tlsRequireTrustedClientCertOnConnect
+          value: "true"
+        command:
+        - sh
+        - -c
+        args: ["bin/apply-config-from-env.py conf/client.conf && bin/apply-config-from-env.py conf/standalone.conf && bin/pulsar standalone -nfw -nss"]
+`
+
+const consumerTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.ConsumerDeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: pulsar-consumer
+spec:
+  selector:
+    matchLabels:
+      app: pulsar-consumer
+  template:
+    metadata:
+      labels:
+        app: pulsar-consumer
+    spec:
+      containers:
+        - name: pulsar-consumer
+          image: ghcr.io/pulsar-sigs/pulsar-client:v0.3.1
+          imagePullPolicy: IfNotPresent
+          readinessProbe:
+            tcpSocket:
+              port: 9494
+          args: ["consumer","--broker","pulsar://{{.StatefulSetName}}.{{.TestNamespace}}:6650","--topic","persistent://public/default/keda","--subscription-name","keda","--consume-time","200"]
+
+`
+
+const scaledObjectTemplate = `
+
+---
```

Comment:
Delete please these lines

Reply:
@liangyuanpeng could you please fix this, so we can merge this PR? We are going to do a new relaese soon. Thanks